core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/promisehandlers/FinalizeCheckpointHandler.scala

```diff
@@ -0,0 +1,59 @@
+package edu.uci.ics.amber.engine.architecture.worker.promisehandlers
+
+import edu.uci.ics.amber.engine.architecture.worker.{
+  DataProcessorRPCHandlerInitializer,
+  WorkflowWorker
+}
+import edu.uci.ics.amber.engine.architecture.worker.WorkflowWorker.MainThreadDelegate
+import edu.uci.ics.amber.engine.architecture.worker.promisehandlers.FinalizeCheckpointHandler.FinalizeCheckpoint
+import edu.uci.ics.amber.engine.common.{CheckpointState, CheckpointSupport, SerializedState}
+import edu.uci.ics.amber.engine.common.rpc.AsyncRPCServer.ControlCommand
+import edu.uci.ics.amber.engine.common.storage.SequentialRecordStorage
+import edu.uci.ics.amber.engine.common.virtualidentity.ChannelMarkerIdentity
+
+import java.net.URI
+import java.util.concurrent.CompletableFuture
+import scala.collection.mutable.ArrayBuffer
+
+object FinalizeCheckpointHandler {
+  final case class FinalizeCheckpoint(checkpointId: ChannelMarkerIdentity, writeTo: URI)
+      extends ControlCommand[Long]
+}
+
+trait FinalizeCheckpointHandler {
+  this: DataProcessorRPCHandlerInitializer =>
+
+  registerHandler { (msg: FinalizeCheckpoint, sender) =>
```

Comment:
the logic is correct but hard to read. add some comments? otherwise we will forget about how it works later.

Reply:
I think the log statements are pretty explainable.